tests/unit/simple_model.py

```diff
@@ -178,13 +178,18 @@ def forward(self, x, y, **kwargs):
         return hidden_dim
 
 
-def random_dataloader(model, total_samples, hidden_dim, device, dtype=torch.half):
-    batch_size = model.train_micro_batch_size_per_gpu()
+def random_dataset(total_samples, hidden_dim, device, dtype=torch.half):
     train_data = torch.randn(total_samples, hidden_dim, device=device, dtype=dtype)
     train_label = torch.empty(total_samples,
                               dtype=torch.long,
                               device=device).random_(hidden_dim)
     train_dataset = torch.utils.data.TensorDataset(train_data, train_label)
+    return train_dataset
+
+
+def random_dataloader(model, total_samples, hidden_dim, device):
```

Comment:
Looks like `dtype=torch.half` was dropped from `random_dataloader()` signature. Please restore.

Reply:
@tjruwase Made the changes. Thanks!

Reply:
@tjruwase The unit test fails again. Says `SimpleModel` has no attribute `device`.

Should I pass in the config, `device = 'cpu'`?

Reply:
Good catch! I just noticed this myself. Please `torch.cuda.curent_device()`.

Reply:
Done!

Reply:
So, it turns out that torch.cuda.current_device() will not work because of a pre-existing problem with our unit tests unrelated to this PR. So in the meantime, I have pushed a change using torch.device('cpu'), as your earlier suggested, to unblock this PR. I will address the other issue in our unit tests later.

In case you are curious, our unit test problem has to do with cuda initialization like this [one](https://github.com/pytorch/pytorch/issues/21092).

Reply:
Okay, CI has completed and all looks good. I am ready to merge whenever you are. Please let me know.

Reply:
@tjruwase Thanks a lot! I am ready to merge.